src/software/new_geom/convex_polygon.h

```diff
@@ -10,13 +10,30 @@ class ConvexPolygon : public Polygon, public ConvexShape
 {
    public:
     ConvexPolygon() = delete;
+
     /**
      * Returns the area of this convex polygon
      *
      * @return The area of this convex polygon
      */
     double area() const override;
 
+    /**
+     * Returns the centroid of this convex polygon
+     *
+     * @return The centroid of this convex polygon
+     */
+    Point centroid() const;
+
+    /**
+     * Returns the convex polygon expanded towards v
```

Comment:
super nit on my preferred wording
```suggestion
     * Returns the convex polygon expanded in the direction of v
```